cli/flox/src/commands/init/python.rs

```diff
@@ -950,69 +931,48 @@ mod tests {
 
         assert_eq!(pyproject.unwrap(), PyProject {
             provided_python_version: ProvidedVersion::Compatible {
-                requested: Some(">=3.8".to_string()),
-                compatible: ProvidedPackage::new("python3", vec!["python39"], "3.9.18"),
+                requested: Some("<=3.10".to_string()),
+                compatible: ProvidedPackage::new("python3", vec!["python3"], PYTHON_310_VERSION),
             },
         });
     }
 
-    /// ProvidedVersion::Incompatible should be returned for requires-python = "1"
+    /// ProvidedVersion::Compatible should be returned for requires-python with space.
     #[tokio::test]
-    async fn pyproject_unavailable_version_with_catalog() {
+    async fn pyproject_available_version_with_space() {
         let (mut flox, _temp_dir_handle) = flox_instance();
 
-        if let Client::Mock(ref mut client) = flox.catalog_client {
-            // Response for python version 1 (resolution failure)
-            client.push_resolve_response(vec![]);
-            // Response for unconstrained python version
-            client.push_resolve_response(vec![resolved_pkg_group_with_dummy_package(
-                "python3_group",
-                &System::from("aarch64-darwin"),
-                "python3",
-                "python3",
-                "3.11.6",
-            )]);
-        }
+        flox.catalog_client = auto_recording_catalog_client("python_lte310_with_space");
 
+        // python docs have a space in the version:
+        // https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires
+        // Expect that version requirement to be parsed and passed on to pkgdb in canonical form.
```

Comment:
pkgdb? 🤔

Reply:
Oops, copied too much of the previous comment.